### DISCUSSION TITLE
Which license would be most appropriate?

### DISCUSSION THREAD

**Discussion by CarlosDerSeher** [CarlosDerSeher/snapclient#30] [Q&A]
As the title suggests I am wrapping my head around this but can't find a good answer myself. Original snapcast is licensed under GPL but this project also contains some parts from espressif licensed differently. So please chip in here if you can.

**Comment by daniel-sousa-me** [comment]
If the original code has GPL, you cannot relicense it.

Are the parts from espressif entangled with your code or are they separate files that you're calling? If it's the latter, you're ok. You probably just need to mention that those parts have X license (it will mostly depend on the terms of their license, not the GPL). If it's the former, it is a modification and it will depend a lot on which license it is. If they use a copyleft license, there's a risk you cannot even use that. If it's not copyleft, you can just relicense the entire thing as GPL.

Can you tell me which other licenses are involved? I'll take a look and try to advise you.

**Reply by CarlosDerSeher** [reply]
I will check as soon as I can. But this could take a while as I need to prepare for an exam for the next two weeks. Thanks for your time.

**Reply by CarlosDerSeher** [reply]
@daniel-sousa-me Ok, so i did a really fast check on the licenses used and it seems there is mostly MIT license in the component folder. Some components are apache V2 and one of them is a submodule. 
Also I use FLAC and opus as git submodule, not sure what's the license on those.

Original snapcast project from badaix is GPL but there isn't really any code from his repo in this client. Only the syncing concept was replicated

**Reply by daniel-sousa-me** [reply]
The submodules are OK with the license. You can usually use external libraries without worrying about it because you're just redistributing the code.

The license copyleft provisions only kick in when you want to distribute modifications (if you edit the original code or base your code on the licensed one). When distributing modifications, you must provide the new code with a compatible license.

**Reply by CarlosDerSeher** [reply]
So the best way would probably be to move all those components to a separate git repo (actually is what espressif does i believe) and pull them in as submodules. Then everything can keep it's license and the snapclient part I , jorgen and the others worked on could be licensed GPL without any issues? Is GPL compatible with apache V2? MIT license shouldn't be a problem at all I believe. Not sure how all those IDF parts come into play but I think it is save to ignore these as most of that is also licensed as MIT i believe. Or do I have to check on these parts too, at least the ones I use in my code?
